### ISSUE TITLE
gcc-7.2.0_3 now producing shared objects instead of executables

### ISSUE THREAD

**Issue by neoklis** [voidlinux/void-packages#8893]
### System

* xuname:  Void 4.13.9_1 x86_64 AuthenticAMD uptodate r
* architecture:  x64_64
* libc:  glibc
* package:  gcc-7.2.0_3

### Expected behavior
Produce executables on compilation
### Actual behavior
Produces a shared library object instead of executable
### Steps to reproduce the behavior
Configure and compile a source package.

Hi all,

Tried to compile some of my own applications from my source code, but gcc produces a shared library object instead of the usual binary executable. The shared object has the name the executable would have had if it was produced. Clang though produces the expected results. This is without me changing anything in the source code, bar fixing some bugs in the C source files.

However,  I used clang-analyzer to detect those bugs and clang-analyzer uses gcc to do the scan-build process, so it may have done something to some gcc or ld config file. It is also possible that the problem was created by the recent -Syu I did before starting the debugging effort. I have tried removing gcc with the -Rf flags and re-installing but no effect.

So now I am not sure what caused this odd behaviour, clang-analyzer or the update.

Any hints gladly welcomed!

Neoklis

**Comment by pullmoll** [comment]
Since gcc-7.2.0 the default is to generate position independent executables and strong stack protection.
You probably want to specify `-fno-PIE` in your `CFLAGS` or `CXXFLAGS` and `-no-pie` in your `LDFLAGS` to get the old behaviour.

Take a look at `common/environment/configure/hardening.sh` for how xbps-src sets the flags depending on `$nopie`.

**Comment by chneukirchen** [comment]
This is not a bug, or do the executables not work?

**Comment by neoklis** [comment]
Now I know the executable works, but I was confused by a coincidence of events: I updated my system first, before installing clang-analyzer to check some of my source code. The update also installed gcc-7.2.0 and since clang-analyzer uses gcc by default, when I started testing I was getting PIE executables (didn't know the new gcc does that by default). And finally my filer (ROX-Filer) was displaying the executables as a shared object and would not start them when clicking on them in the filer window. So I wasn't sure what was happening there - only later I tried running the executables from an xterm, which worked.

Apologies for a not very constructive post!

Neoklis
